tests/Tokenizer/TokensAnalyzerTest.php

```diff
@@ -769,246 +769,247 @@ public function testIsLambdaInvalid()
+            yield [
+                '<?php interface Foo { public function bar(): ?\Baz; }',
+                [3 => false, 11 => false, 18 => false],
+            ];
         }
-    }
 
-    public function provideIsConstantInvocation70Cases()
-    {
-        return [
-            [
-                '<?php function x(): FOO {}',
-                [8 => false],
-            ],
-            [
-                '<?php use X\Y\{FOO, BAR as BAR2, BAZ};',
-                [8 => false, 11 => false, 15 => false, 18 => false],
-            ],
-        ];
+        if (\PHP_VERSION_ID >= 80000) {
+            yield [
+                '<?php $a?->b?->c;',
+                [3 => false, 5 => false],
+            ];
+            yield [
+                '<?php try {} catch (Exception) {}',
+                [9 => false],
+            ];
+            yield [
+                '<?php try {} catch (\Exception) {}',
+                [10 => false],
+            ];
+            yield [
+                '<?php try {} catch (Foo | Bar) {}',
+                [9 => false, 13 => false],
+            ];
+        }
     }
 
     /**
      * @param string $source
      *
-     * @dataProvider provideIsConstantInvocation71Cases
-     * @requires PHP 7.1
+     * @dataProvider provideIsConstantInvocationPhp80Cases
+     * @requires PHP 8.0
      */
-    public function testIsConstantInvocation71($source, array $expected)
+    public function testIsConstantInvocationPhp80($source, array $expected)
     {
-        $tokensAnalyzer = new TokensAnalyzer(Tokens::fromCode($source));
-
-        foreach ($expected as $index => $expectedValue) {
-            static::assertSame($expectedValue, $tokensAnalyzer->isConstantInvocation($index), 'Token at index '.$index.' should match the expected value.');
-        }
+        $this->doIsConstantInvocationTest($source, $expected);
     }
 
-    public function provideIsConstantInvocation71Cases()
+    public function provideIsConstantInvocationPhp80Cases()
     {
-        return [
-            [
-                '<?php function x(?FOO $foo) {}',
-                [6 => false],
-            ],
-            [
-                '<?php function x(): ?FOO {}',
-                [9 => false],
-            ],
-            [
-                '<?php try {} catch (FOO|BAR|BAZ $e) {}',
-                [9 => false, 11 => false, 13 => false],
-            ],
-            [
-                '<?php interface Foo { public function bar(): Baz; }',
-                [16 => false],
-            ],
-            [
-                '<?php interface Foo { public function bar(): \Baz; }',
-                [17 => false],
-            ],
-            [
-                '<?php interface Foo { public function bar(): ?Baz; }',
-                [17 => false],
-            ],
-            [
-                '<?php interface Foo { public function bar(): ?\Baz; }',
-                [18 => false],
-            ],
+        yield [
+            '<?php #[Foo] function foo() {}',
```

Comment:
please add `'<?php #[\Foo] function foo() {}',` , `'<?php #[\A\Foo] function foo() {}'` and `'<?php #[A\Foo] function foo() {}',`